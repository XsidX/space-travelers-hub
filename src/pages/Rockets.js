import React from 'react';

const Rockets = () => (
  <div>
    <h1>Rockets</h1>
  </div>
);

export default Rockets;
