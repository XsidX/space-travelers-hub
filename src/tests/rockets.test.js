import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import * as reactRedux from 'react-redux';
import Rockets from '../components/Rocket';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

describe('Rockets component', () => {
  const { useSelector } = reactRedux;
  const { useDispatch } = reactRedux;

  beforeEach(() => {
    useSelector.mockImplementation((selector) => selector({
      rockets: {
        rockets: [
          {
            id: '1',
            rocket_name: 'Falcon 1',
            description:
                'Falcon 1 is a partially reusable expendable rocket designed and manufactured by SpaceX.',
            flickr_image:
                'https://farm1.staticflickr.com/813/29188622681_f8f5e7e0e0_o.jpg',
            reserved: false,
          },
        ],
      },
    }));
    useDispatch.mockImplementation(() => jest.fn());
  });

  afterEach(() => {
    useSelector.mockClear();
    useDispatch.mockClear();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<Rockets />);

    expect(tree).toMatchSnapshot();
  });

  it('does not render reserved message if the reserved state is false', () => {
    render(<Rockets />);

    const reservedMessage = screen.queryByText('Reserved', { exact: false });

    expect(reservedMessage).toBeNull();
  });

  it('renders reserved message when the reserved state is true', () => {
    useSelector.mockImplementation((selector) => selector({
      rockets: {
        rockets: [
          {
            id: '1',
            rocket_name: 'Falcon 1',
            description:
                'Falcon 1 is a partially reusable expendable rocket designed and manufactured by SpaceX.',
            flickr_image:
                'https://farm1.staticflickr.com/813/29188622681_f8f5e7e0e0_o.jpg',
            reserved: true,
          },
        ],
      },
    }));
    render(<Rockets />);

    const reservedMessage = screen.getByText('Reserved', { exact: false });

    expect(reservedMessage).toBeInTheDocument();
  });
});
