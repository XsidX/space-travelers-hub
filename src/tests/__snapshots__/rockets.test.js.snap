// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets component renders correctly 1`] = `
<div
  className="rocket-container"
>
  <div
    className="rocket-img"
  >
    <img
      alt="rocket"
      src="https://farm1.staticflickr.com/813/29188622681_f8f5e7e0e0_o.jpg"
    />
  </div>
  <div
    className="rocket-infos"
  >
    <h2>
      Falcon 1
    </h2>
    <p>
      Falcon 1 is a partially reusable expendable rocket designed and manufactured by SpaceX.
    </p>
    <button
      className="action-btn"
      id="1"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
